### PR TITLE
+redis.io

### PR DESCRIPTION
**Pull request by getneil**

### CODE REVIEW

File: projects/redis.io/package.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - bin/redis-server
+  - bin/redis-cli
+  - bin/redis-benchmark
+
+dependencies:
+  openssl.org: ^1
+
+build:
+  dependencies:
+    gnu.org/make: '>=4.3'
```
Comment (jhheider) [inline projects/redis.io/package.yml]: Does it require GMake specifically, or is `tea.xyz/gx/make: '*'` sufficient?

File: projects/redis.io/package.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    freedesktop.org/pkg-config: '>=0.29'
+    tea.xyz/gx/cc: c99
+  script:
+    PREFIX={{prefix}}
+    BUILD_TLS=yes
+    make install
```
Comment (jhheider) [inline projects/redis.io/package.yml]: I'd change this to:
```yaml
script: make install
env:
  PREFIX: ${{prefix}}
  BUILD_TLS: yes
```

File: projects/redis.io/package.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    # TODO: configure default redis.conf if necessary
+test:
+  script: |
+    redis-server --test-memory 2
+    redis-cli --version
+    redis-benchmark --version
```
Comment (jhheider) [inline projects/redis.io/package.yml]: Good number of tests. Two notes:

- if `script:` is the only subkey, you can pass the script directly on the `test:` key
- is it worth spinning up the server, running a couple of set/gets then spinning it down, to be sure it's working?

**Review by jhheider — APPROVED** [review]
Looks good. Mostly just style questions/comments. Eventually, we'll want some kind of service runner for `tea` bins, but no need to hold up packaging for it (indeed, having some servers packaged should make it easier to write/test).

File: projects/redis.io/package.yml (hunk 3 above)
Comment (getneil) [reply projects/redis.io/package.yml]: added.

File: projects/redis.io/package.yml (hunk 2 above)
Comment (getneil) [reply projects/redis.io/package.yml]: thanks

File: projects/redis.io/package.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    echo $redis_value
+
+    redis-cli shutdown
+
+    if [ $redis_value = "value123" ]; then
+      echo "redis works!"
+    else
+      echo "redis install doesnt work!"
+      set -e
+    fi
```
Comment (jhheider) [inline projects/redis.io/package.yml]: simpler exit condition might just be `test "$redis_value" = "value123"` and let the error code tell the story.

Comment (jhheider) [reply projects/redis.io/package.yml]: or, even just replace the latter half with `test "$(redis-cli --raw GET key123)" = "value123"`

**Comment by jhheider** [comment]
I'll fix that build script error :/

File: projects/redis.io/package.yml (hunk 1 above)
Comment (getneil) [reply projects/redis.io/package.yml]: tested replacement and it works

File: projects/redis.io/package.yml (hunk 4 above)
Comment (getneil) [reply projects/redis.io/package.yml]: that is much cleaner. added it.

**Comment by mxcl** [comment]
lovely job here. The test especially is magnificent. Good understanding of tea. I wonder about the cores key in build tho. Is that something we added?

**Comment by jhheider** [comment]
Yes, for larger requirements: https://github.com/teaxyz/pantry.zero/wiki/Structure-of-a-package.yml#cores

That note should say it defaults to 2 if omitted...

**Comment by jhheider** [comment]
Wiki updated.

**Pull request merged**